core-meta.html

```diff
@@ -87,20 +79,29 @@
          * @type String
          * @default ''
          */
-        key: String
-      },
-
-      bind: {
-        value: 'valueChanged',
-        key: 'keyChanged'
-      },
+        key: {
+          type: String,
+          value: '',
+          notify: true,
+          observer: 'keyChanged'
+        },
 
-      configure: function () {
-        return {
-          type: 'default',
-          key: '',
-          value: this
-        };
+        /**
+         * The value of the meta-data to store. If no value is specified, the
+         * value defualts to a self reference (`this`).
+         *
+         * @attribute value
+         * @type *
+         * @default this
+         */
+        value: {
+          type: String,
```

Comment:
If you don't put a type, it doesn't take the value from the attribute. Type of `String` is required to prevent weird serialization. I may just be missing something important here though, because none of this was expected behavior for me.